### PR TITLE
Ees 2687 extract release approval status service

### PR DESCRIPTION
**Pull request by duncan-at-hiveit**
This PR:
- extracts elements of ReleaseService around approval into ReleaseApprovalService
- extracts elements of MethodologyService around approval into MethodologyApprovalService 

Also broke out EES-2747 to cover a final piece of refactoring that I ran out of time to do!  Referenced it in the code.

image

### CODE REVIEW

File: src/GovUk.Education.ExploreEducationStatistics.Admin/Controllers/Api/ReleasesController.cs [hunk 1]
```diff
@@ -177,8 +181,9 @@ public async Task<ActionResult<List<ReleaseStatusViewModel>>> GetReleaseStatuses
         public async Task<ActionResult<ReleaseViewModel>> CreateReleaseStatus(ReleaseStatusCreateViewModel request,
             Guid releaseId)
         {
-            return await _releaseService
+            return await _releaseApprovalService
                 .CreateReleaseStatus(releaseId, request)
+                .OnSuccess(_ => _releaseService.GetRelease(releaseId))
                 .HandleFailuresOrOk();
```
[the remark this answers is not in the document]
Comment (duncan-at-hiveit) [reply src/GovUk.Education.ExploreEducationStatistics.Admin/Controllers/Api/ReleasesController.cs:187]: Yeah, looks like `ReleaseStatusEditPage.tsx` uses the response.  Good suggestion though :)

File: src/GovUk.Education.ExploreEducationStatistics.Admin/Services/Interfaces/IReleaseApprovalService.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
+using System;
```
[the remark this answers is not in the document]
Comment (duncan-at-hiveit) [reply src/GovUk.Education.ExploreEducationStatistics.Admin/Services/Interfaces/IReleaseApprovalService.cs:1]: Done here.

File: src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseApprovalService.cs [hunk 3]
```diff
@@ -0,0 +1,188 @@
+using System;
```
[the remark this answers is not in the document]
Comment (duncan-at-hiveit) [reply src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseApprovalService.cs:1]: And done here too :)

File: src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseApprovalService.cs [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+        private readonly IPersistenceHelper<ContentDbContext> _persistenceHelper;
+        private readonly IUserService _userService;
+        private readonly IPublishingService _publishingService;
+        private readonly IReleaseChecklistService _releaseChecklistService;
+        private readonly IContentService _contentService;
+        private readonly IPreReleaseUserService _preReleaseUserService;
+        private readonly IReleaseFileRepository _releaseFileRepository;
+        private readonly IReleaseFileService _releaseFileService;
+
+        public ReleaseApprovalService(ContentDbContext context, IPersistenceHelper<ContentDbContext> persistenceHelper, IUserService userService, IPublishingService publishingService, IReleaseChecklistService releaseChecklistService, IContentService contentService, IPreReleaseUserService preReleaseUserService, IReleaseFileRepository releaseFileRepository, IReleaseFileService releaseFileService)
```
[the remark this answers is not in the document]
Comment (duncan-at-hiveit) [reply src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseApprovalService.cs]: Wow yes.  Not sure how that snuck in there!  Reformatted.

**Review by benoutram — APPROVED** [review]

**Pull request merged**